core/src/main/scala/org/bykn/bosatsu/Declaration.scala

```diff
@@ -368,6 +444,10 @@ object Declaration {
    * Foo { bar: 12 }
    */
   case class RecordConstructor(cons: Constructor, arg: NonEmptyList[RecordArg])(implicit val region: Region) extends NonBinding
+  /**
+   * This represents interpolated strings
+   */
+  case class StringDecl(items: NonEmptyList[Either[NonBinding, (Region, String)]])(implicit val region: Region) extends NonBinding
```

Comment:
we have been using the trick that extra param groups on case classes don't count on `equals`. Embedding this region here on the string complicates things quite a bit...

Of course, maybe reusing Literal would sidestep it, but in any case I added replaceRegion to help make equality ignoring regions.